modules/gistic2/1.0/gistic2.smk

```diff
@@ -0,0 +1,304 @@
+        refgene_mat = CFG["dirs"]["inputs"] + "references/{projection}.refgene.mat"
+    group: 
+        "input_and_format"
+    params:
+        url = "http://bcgsc.ca/downloads/morinlab/gistic2_references/{projection}.refgene.mat",
+        folder = CFG["dirs"]["inputs"] + "references"
+    shell:
+        op.as_one_line("""
+        wget -P {params.folder} {params.url} 
+        """)
+
+def _get_seg_input_params(wildcards, input_dir=CFG["dirs"]["inputs"], seq_types = CFG["samples"]["seq_type"].unique()):
+    if "capture" in seq_types and "genome" in seq_types:
+        param = "--genome " + input_dir + "genome--projection/all--" + wildcards.projection + ".seg --capture " \
+                + input_dir + "capture--projection/all--" + wildcards.projection + ".seg"
+    elif "capture" in seq_types and "genome" not in seq_types:
+        param = "--capture " + input_dir + "capture--projection/all--" + wildcards.projection + ".seg"
+    elif "capture" not in seq_types and "genome" in seq_types:
+        param = "--genome " + input_dir + "genome--projection/all--" + wildcards.projection + ".seg"
+    
+    return(param)
+
+# Merges capture and genome seg files if available, and subset to the case_set provided
+rule _gistic2_prepare_seg:
+    input:
+        seg = expand(str(rules._gistic2_input_seg.output.seg),
+                    allow_missing=True,
+                    seq_type=CFG["samples"]["seq_type"].unique()
+                    ),
+        all_sample_sets = ancient(str(rules._gistic2_input_sample_sets.output.all_sample_sets)),
+        seg_dir = CFG["dirs"]["inputs"]
+    output:
+        seg = CFG["dirs"]["prepare_seg"] + "{case_set}--{projection}.seg"
+    log:
+        stdout = CFG["logs"]["prepare_seg"] + "{case_set}--{projection}.stdout.log",
+        stderr = CFG["logs"]["prepare_seg"] + "{case_set}--{projection}.stderr.log"
+    group: 
+        "input_and_format"
+    params:
+        script = CFG["prepare_seg"],
+        case_set = CFG["case_set"],
+        seg_input_params = _get_seg_input_params
+    shell:
+        op.as_one_line("""
+        Rscript {params.script} 
+        {params.seg_input_params} 
+        --projection {wildcards.projection} 
+        --output_dir $(dirname {output.seg})/ 
+        --all_sample_sets {input.all_sample_sets} 
+        --case_set {params.case_set} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+# Create a markers file that has every segment start and end that appears in the seg file
+rule _gistic2_make_markers:
+    input:
+        seg = str(rules._gistic2_prepare_seg.output.seg)
+    output:
+        temp_markers = temp(CFG["dirs"]["markers"] + "temp_markers--{case_set}--{projection}.txt"),
+        markers = CFG["dirs"]["markers"] + "markers--{case_set}--{projection}.txt"
+    log:
+        stderr = CFG["logs"]["markers"] + "{case_set}--{projection}--markers.stderr.log"
+    shell: 
+        op.as_one_line("""
+        sed '1d' {input.seg} | cut -f2,3 > {output.temp_markers}
```

Comment:
I don't remember if this has to deal with the header of the seg file or assumes there is no header - I trust this works as expected because I know you run this successfully already!

Reply:
Yes, the `'1d'` tells sed to skip the first line